Source/com/drew/imaging/quicktime/QuickTimeReader.java

```diff
@@ -47,7 +47,7 @@ public static void extract(@NotNull InputStream inputStream, @NotNull QuickTimeH
     private static void processAtoms(StreamReader reader, long atomEnd, QuickTimeHandler handler)
     {
         try {
-            while (atomEnd == -1 || reader.getPosition() < atomEnd) {
+            while (reader.available() > 0 && (atomEnd == -1 || reader.getPosition() < atomEnd)) {
```

Comment:
What's the rationale behind adding the `available` check? I don't think this does what you want it to do.

Reply:
I experienced a read beyond EOF, because of Canon additional 0-byte in Canon, that is the reason.

Reply:
When `available()` returns zero it doesn't necessarily mean the stream has completed. It just means there is data that can be read without blocking the calling thread. With this change, it's possible the loop exits even though data might become available later.

https://docs.oracle.com/javase/7/docs/api/java/io/InputStream.html#available()
https://stackoverflow.com/q/3695372/24874

We'll need a different fix to work around reading after EOF unfortunately.

Reply:
reader.available() removed

Reply:
Thanks for following up. I'll get this merged ASAP.